packages/wallet/ui/src/components/ConnectionSettingsDialog.jsx

```diff
@@ -173,6 +174,17 @@ const ConnectionSettingsDialog = ({
   }
 
   const saveAndClose = () => {
+    if (
+      JSON.stringify(config) === JSON.stringify(connectionConfig) &&
```

Comment:
is this to do a deep equal or is there something about JSON specifically? don't we have a deep equal util somewhere?

Reply:
Yes it just happens that in this case JSON.stringify suffices. We don't have one in the frontend (outside of the test env).

I just went ahead and dropped a util function in rather than importing a library to do it.